src/Analyzers/Core/Analyzers/AbstractBuiltInCodeStyleDiagnosticAnalyzer_Core.cs

```diff
@@ -10,13 +10,8 @@ namespace Microsoft.CodeAnalysis.CodeStyle
 {
     internal abstract partial class AbstractBuiltInCodeStyleDiagnosticAnalyzer : DiagnosticAnalyzer, IBuiltInAnalyzer
     {
-        protected readonly string? DescriptorId;
-
         protected readonly DiagnosticDescriptor Descriptor;
```

Comment:
I'm not comfortable with this field too. But removing it will require more changes. So keeping it for now.

Reply:
That's fair.  might want to rename it to FirstDescriptor.

Reply:
@CyrusNajmabadi Do you prefer that to be in this PR?